### PR TITLE
 instructions to run sample code with mongobetween  proxy

### PR DESCRIPTION
**Pull request by sourav11b**
*Issue #, if available:*

*Description of changes:*

adding instructions to run sample code with mongobetween  proxy

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: samples/mongobetween_sample/README.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+## Introduction
+
+### Why do we need a connection proxy for Amazon DocumentDB
+
+The maximum number of connections that DocumentDB currently supports is 30K per instance, and this is a hard limit. Scenarios where we see customers hitting these limits are with applications that are deployed using containers and have connections that are opened for a prolonged period of time. A common design solution to minimize the connections created on the database instances is to use a connection proxy layer. In this article, we introduce an open-source proxy solution and provide steps to install and deploy this proxy, along with the required application changes.
+
+#### What is Mongobetween?
+
+Mongobetween is a lightweight MongoDB connection pooler written in Golang. Its primary function is to handle a large number of incoming connections and multiplex them across a smaller connection pool to one or more DocumentDB clusters.
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:11]: I don't understand the second sentence, how/why would you use it for >1 DocumentDB clusters?

File: samples/mongobetween_sample/README.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+More resources:
+    
+1.  [Github code repo](https://github.com/coinbase/mongobetween/tree/master) 
+2.  [Coinbase blog](https://www.coinbase.com/blog/scaling-connections-with-ruby-and-mongodb)
+
+## Prerequisites
+
+To implement this solution, you must have the following prerequisites:
+    
+* An [Amazon EC2 Instance](https://aws.amazon.com/pm/ec2/) where you can run Python scripts to generate database connections. You can use an existing Amazon EC2 instance or [create a new one](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EC2_GetStarted.html). Use a c5.18xlarge instance type for this test so that we have enough CPU for effective multi-processing. Configure appropriate network settings so that you are able to SSH into the instance.
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:22]: Did you mean "c5.18xlarge"?

File: samples/mongobetween_sample/README.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+* An Amazon DocumentDB cluster with at least two db.r6g.large instances. You can use an existing Amazon DocumentDB cluster or [create a new one](https://docs.aws.amazon.com/documentdb/latest/developerguide/db-cluster-create.html). This post assumes the default values for port (27017) and TLS (enabled) settings.
+
+* A security group that enables you to connect to your Amazon DocumentDB cluster from your Amazon EC2 instance. You can use an existing security group or [create a new one](https://docs.aws.amazon.com/documentdb/latest/developerguide/get-started-guide.html#cloud9-security). You may also use the [Connect using Amazon EC2](https://docs.aws.amazon.com/documentdb/latest/developerguide/connect-ec2.html) feature to connect your Amazon DocumentDB cluster to your Amazon EC2 instance. 
+
+* Python 3+. Modules pymongo
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:28]: Be more specific about Python version, 3.x+ (what is your value for x)

File: samples/mongobetween_sample/README.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    -enable-sdam-logging
+            Enable SDAM(Server Discovery And Monitoring) logging
+    -enable-sdam-metrics
+            Enable SDAM(Server Discovery And Monitoring) metrics
+    -loglevel string
+            One of: debug, info, warn, error, dpanic, panic, fatal (default "info")
+    -network string
+            One of: tcp, tcp4, tcp6, unix or unixpacket (default "tcp4")
+    -password string
+            MongoDB password
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:75]: Do you mean DocumentDB?

File: samples/mongobetween_sample/README.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    -ping
+            Ping downstream MongoDB before listening
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:77]: Do you mean DocumentDB?

File: samples/mongobetween_sample/README.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    -pretty
+            Pretty print logging
+    -statsd string
+            Statsd address (default "localhost:8125")
+    -unlink
+            Unlink existing unix sockets before listening
+    -username string
+            MongoDB username
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:85]: DocumentDB?

File: samples/mongobetween_sample/README.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+        mkdir working_directory
+        cd working_directory
+
+2. Create a log directory for the output of the script.
+
+        mkdir logs
+
+3. Copy the file ```test_mongobetween.py``` to the working directory. This file contains Python code that spins off a number of processes, as provided by the argument ```--mongo-clients-count```. Each process creates its own collection, inserts 1000 documents into the primary instance of Amazon DocumentDB, as provided by the argument ```--docdb-uri```, and then reads the 1000 documents back from a replica.
+
+3. Download the file ```global-bundle.pem``` in the working directory.
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:101]: Pem is optional (if cluster is using encryption)

File: samples/mongobetween_sample/README.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+
+Create a dashboard and [view/edit source](https://repost.aws/questions/QUzarTQVN5TgqAVeXR6_h6Bg/cloudwatch-dashboard-source-json). Copy the contents of the previous step and save the dashboard.
+
+![dashboard-screenshot](files/dashboard-screenshot.png)
+
+Note: This dashboard was created for a cluster with a topology of one primary and one replica instance. Change the content according to your cluster topology.
+
+## Conclusion
+
+If you can use Mongobetween effectively with your containerized applications, this will let you handle unplanned scaling events gracefully without exceptions being generated in the application. Please note that whenever you are sending more connections to Mongobetween than what is established to the DocumentDB cluster from MongoBetwen, you will notice increased latencies in the application. This is a trade-off that you need to consider in your application design, timeout settings, and exception handling.
```
Comment (tmcallaghan) [inline samples/mongobetween_sample/README.md:238]: Typo "MongoBetwn"

File: samples/mongobetween_sample/README.md (hunk 6 above)
Comment (sourav11b) [reply samples/mongobetween_sample/README.md:85]: this is message mongobetween prints. to change this we need to change mongobetween code

File: samples/mongobetween_sample/README.md (hunk 1 above)
Comment (sourav11b) [reply samples/mongobetween_sample/README.md:11]: mongobetween can also act as a proxy between multiple documentdb clusters. this use case is not covered here

**Review by tmcallaghan — APPROVED** [review]
Approved.

**Pull request merged**